contracts/misc/ETHWithdrawal.sol

```diff
@@ -0,0 +1,223 @@
+error AlreadyMinted();
+error NotMature();
+error InvalidParams();
+
+contract ETHWithdrawal is
+    Initializable,
+    ReentrancyGuard,
+    AccessControl,
+    ERC1155,
+    IERC721Receiver,
+    IETHWithdrawal
+{
+    using SafeERC20 for IERC20;
+    using WadRayMath for uint256;
+
+    bytes32 public constant DEFAULT_ISSUER_ROLE = keccak256("DEFAULT_ISSUER");
+
+    mapping(uint256 => IETHWithdrawal.TokenInfo) private tokenInfos;
+    mapping(IETHWithdrawal.StakingProvider => address)
+        public providerStrategyAddress;
+
+    uint256 public nextTokenId;
+
+    constructor(string memory uri_) ERC1155(uri_) {}
+
+    function initialize(address _admin) public initializer {
+        require(_admin != address(0), "Address cannot be zero");
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(DEFAULT_ISSUER_ROLE, _admin);
+    }
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(AccessControl, ERC1155)
+        returns (bool)
+    {
+        return
+            interfaceId == type(IETHWithdrawal).interfaceId ||
+            interfaceId == type(IERC721Receiver).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc IETHWithdrawal
+    function mint(
+        IETHWithdrawal.StakingProvider provider,
+        uint64 exitEpoch,
+        uint64 withdrawableEpoch,
+        uint256 balance,
+        address recipient,
+        uint256 withdrawableTime
+    )
+        external
+        nonReentrant
+        onlyRole(DEFAULT_ISSUER_ROLE)
+        returns (uint256 tokenId)
+    {
+        if (provider == IETHWithdrawal.StakingProvider.Validator) {
+            if (block.timestamp >= withdrawableTime) {
+                revert AlreadyMature();
+            }
+
+            if (recipient == address(0) || balance == 0) {
+                revert InvalidParams();
+            }
+
+            tokenId = nextTokenId++;
+
+            if (tokenInfos[tokenId].balance > 0) {
+                revert AlreadyMinted();
+            }
+
+            tokenInfos[tokenId] = IETHWithdrawal.TokenInfo(
+                provider,
+                exitEpoch,
+                withdrawableEpoch,
+                balance,
+                withdrawableTime
+            );
+            _mint(recipient, tokenId, balance, bytes(""));
+            emit Mint(recipient, tokenId, balance);
+        } else {
+            revert Unimplemented();
+        }
+    }
+
+    /// @inheritdoc IETHWithdrawal
+    function burn(
+        uint256 tokenId,
+        address recipient,
+        uint256 amount
+    ) external nonReentrant {
+        TokenInfo memory tokenInfo = tokenInfos[tokenId];
+        if (tokenInfo.provider == IETHWithdrawal.StakingProvider.Validator) {
+            if (block.timestamp < tokenInfo.withdrawableTime) {
+                revert NotMature();
+            }
+
+            Helpers.safeTransferETH(recipient, tokenInfo.balance);
```

Comment:
Helpers.safeTransferETH(recipient, amount) ?

Reply:
emmm you are right